### DISCUSSION TITLE
How to use gopro-join

### DISCUSSION THREAD

**Discussion by DougEdwards47** [time4tea/gopro-dashboard-overlay#194] [Q&A]
I just want to join a series of GoPro files (caused by the automatic splitting of long recordings by GoPro) without losing any of the embedded metadata. But I just can't get gopro-join to work. It is stated that "the first argument is a single MP4 file from the sequence". What is the sequence? - what is meant by "the sequence" ? How do I specify it or point to its location? Does it matter where in the "sequence", the file is located  and how might "order" be defined: alphabetic order, file creation order, first time stamp in the embedded gpx data,  last time stamp in the embedded gpx data, etc?

**Comment by time4tea** [comment]
hi there - thanks for your interest in the software! - Sorry you've been having trouble. I should say up front that the joining method isn't perfect. The reason being that the gopro data and the "surrounding" MP4 file are kind of interlinked, and the joining method doesn't quite take that into account.
You should be able to invoke the program just as described at  https://github.com/time4tea/gopro-dashboard-overlay/tree/main/docs/bin#gopro-joinpy ... just give it the name of *one* of the input files, and it will join them all to the output file.

Did that not work? Please can you let me know the names of the input files, and what command line you tried. What GoPro model is this? I have heard that Hero12 has some new features, but I've not got one, so I haven't seen.

Thanks!

**Comment by DougEdwards47** [comment]
Thanks for your prompt reply.
Short response: I have solved my problem. Slightly longer explanation below.

 I would have responded sooner, but I have been verifying the file numbering algorithm used by GoPro. It doesn’t help that some of the historic information on the web is of date. A current guide seems to be: https://tinyurl.com/GoPro-File-Names

I have now worked out what is going on.  My problem was understanding the wording of the description of the positional arguments:
	"input	A single MP4 file from the sequence”
I now understand that the sequence referred to is the sequence of files caused by splitting large video recordings into what GoPro refers to as Chapters. e.g. GX010052, GX,020052, GX030052. This makes sense in retrospect.

It should be noted that the algorithm does not work with the older (bizarre) numbering scheme e.g. that used by my GoPro Hero Session. The first file in the sequence of GOPR1460, GP011460, GP021460, GP031460 is (understandably) omitted in the join. This doesn’t bother me though since I no longer use the Hero.

However, what I also wanted to do was to join non-time-contiguous clips together. For example, clip1 might be an activity before lunch. I switch off my camera, then start it again after lunch to record clip 2 etc. I would then get a series of clips: GX010051, GX010052. Not really a problem, I could fake the clip numbering - but that results in an output file with incorrect SMPTE time codes on the 2nd and subsequent clips or I could join them in a conventional video editor like iMovie.  Again, now I understand what is going on, I can live with it (once I decide what I really want).

I have played with GoPro Quik and Garmin VIRB EDIT on my desktop Mac and found both lacking in irritating respects. GoPro Quik now works (at last) on MacOS, it is simple to use, but the widgets / gauges are ugly and very inflexible. Garmin VIRB EDIT works, with a lot of control over the widgets/gauges, but doesn’t accept HEVC files and recoding HEVC files to H264 in Quicktime loses the meta-data, and requires a change in suffix in the produced file from .mov to .mp4. Not sure why that works - I thought they denoted different container formats. Anyway, it is all solvable, but tedious.

So, very pleased to have come across gopro-dashboard-overlay.  It seems to work out of the box, so to speak, although finding my way around the configurations is somewhat complex.

Some contextual information, I have a GoPro 11. When upgrading from a GoPro Hero Session, I bought the GoPro 11 model rather than the GoPro12 because the latter didn’t have built-in GPS.  I am running the software on an 2018 Intel Mac Mini (3.2 GHz, 6-core i7) running Sonoma 14.4.1

— 
Doug Edwards
M20 3LE
07966 049908

**Comment by acrobaticparticle** [comment]
Hello.
I just wanted to add some info for those who might suffer from the same problem.
I wanted to join mp4 video files created with my gopro 8. The files in the sequence were not auto split but I recorded them one by one. When I wanted to use gopro join it could not find the the rest of the files in the same folder only the first I give. For me the file pattern looks like GH[n2][n4].mp4. So two fixed letter then 2 digits then 4 digits. 
I do not know about how many variations can be here and how the different versions of gopro cameras generate the filenames but for me the checker regular expression in the code is unnecessarily complex for this so I changed a bit.
filenaming.py
    def related_files(self, d: Path, listdir=os.listdir):
        find = re.compile(r"G{l}\d+\.{e}".format(
            l=self.letter,
            e=self.extension
        ))
        
Another side note: The latest version of gopro dashboard overlay (126) does not work on windows again because of the missing gpmd module There is a gpmf instead which I could used so I changed the import to that in gopro-dasboard.py and it worked again.
